### DISCUSSION TITLE
How root is disabled in Android?

### DISCUSSION THREAD

**Discussion by korpe1** [termux/termux-app#3851] [Q&A]
How exactly root is disabled from user in Android? If I want make same for Ubuntu, what I have to do? Of course Android isn't ordinary Linux and Google could have just rewrite all related to it in the kernel, but I doubt they did.

What Magisk and others exactly do?

Thank you :)

---
_This isn't directly only Termux related question, but here is a lot Android devs that are familiar with ordinary environment of Linux OS family, Linux kernel and command line. I hope it's okay to you, but be free to delete this if it isn't._

**Comment by korpe1** [comment]
Any devs can tell?

**Reply by twaik** [reply]
Tell what exactly? How root works or does not work on Android?

**Reply by twaik** [reply]
Android does not have builtin su or sudo, so it is not possible to elevate current user privilegies after dropping them by zygote. Zygote starts processes in needed user sandbox on demand. It is simple.
